### PR TITLE
feat: implement multiple sort field when list applications

### PR DESCRIPTION
**Pull request by aneojgurhem**
This PR implements multiple sort field when list applications for GUI.

fix: https://github.com/aneoconsulting/ArmoniK/issues/896

### CODE REVIEW

File: Adaptors/MongoDB/src/IMongoQueryableExt.cs [hunk 1]
```diff
@@ -40,4 +43,36 @@ internal static class IMongoQueryableExt
       }
     }
   }
+
+  public static IOrderedMongoQueryable<T> OrderByList<T>(this IMongoQueryable<T>            queryable,
+                                                         List<Expression<Func<T, object?>>> orderFields,
```
Comment (lemaitre-aneo) [inline Adaptors/MongoDB/src/IMongoQueryableExt.cs:48]:
```suggestion
                                                         ICollection<Expression<Func<T, object?>>> orderFields,
```
I think there is no need for a List specifically. You could at least take a `ICollection`, and maybe a `IEnumerable` even.

File: Adaptors/MongoDB/src/IMongoQueryableExt.cs [hunk 2]
```diff
@@ -40,4 +43,36 @@ internal static class IMongoQueryableExt
...
+                                                         bool                               ascOrder = true)
```
Comment (lemaitre-aneo) [inline Adaptors/MongoDB/src/IMongoQueryableExt.cs:49]: Order should most likely be per "column", and not necessarily the same for all "columns".

File: Common/src/Utils/IQueryableExt.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace ArmoniK.Core.Common.Utils;
+
+public static class IQueryableExt
+{
+  public static IOrderedQueryable<T> OrderByList<T>(this IQueryable<T>                 queryable,
```
Comment (lemaitre-aneo) [inline Common/src/Utils/IQueryableExt.cs:27]: Do you really have to implement it twice?

File: Adaptors/MongoDB/src/IMongoQueryableExt.cs (hunk 1 above)
Comment (aneojgurhem) [reply Adaptors/MongoDB/src/IMongoQueryableExt.cs:48]: ICollection is effectively good. We will have a multiple iteration problem with IEnumerable due to the first and skip.

File: Common/src/Utils/IQueryableExt.cs (hunk 3 above)
Comment (aneojgurhem) [reply Common/src/Utils/IQueryableExt.cs:27]: due to https://github.com/aneoconsulting/ArmoniK.Core/blob/13805d638c8c0162c88070c6649ae53aba8934dc/Adaptors/MongoDB/src/TaskTable.cs#L473
that we created, we cannot change the type of the output of the orderby. For now, I do not have a better solution

File: Adaptors/MongoDB/src/IMongoQueryableExt.cs (hunk 2 above)
Comment (aneojgurhem) [reply Adaptors/MongoDB/src/IMongoQueryableExt.cs:49]: It is not needed and simplifies code for now

**Review by lemaitre-aneo — COMMENTED** [review]
It would be nice to test if the sort actually works.

File: Common/tests/TestBase/TaskTableTestBase.cs [hunk 4]
```diff
@@ -1250,6 +1254,151 @@ public async Task ListApplicationFromTasksShouldSucceed()
...
+                                     .ConfigureAwait(false);
+
+      var listTasksResponseTaskData = listTasks.applications.ToList();
+      foreach (var task in listTasksResponseTaskData)
+      {
+        Console.WriteLine(task);
+      }
+
+      // app2 before app1 because desc ordering in request
+      // app2 before app3 because desc and ordering by version then by service
```
Comment (lemaitre-aneo) [inline Common/tests/TestBase/TaskTableTestBase.cs:1391]: From the comment, we cannot say which app1 or app3 should be first.

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**